File: backend/src/main/kotlin/com/example/backend/models/LigneCaisse.kt
package com.example.backend.models

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "ligne_caisse")
class LigneCaisse {
   @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "caisse_id")
  var caisse: Caisse? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "produit_id")
  var produit: com.example.backend.models.Produit? = null

  @Column(name = "libelle", nullable = false)
  var libelle: String? = null

  @Column(name = "dateLigne", nullable = false)
  var dateLigne: LocalDateTime? = null

  @Column(name = "debit")
  var debit: Double? = null

  @Column(name = "credit")
  var credit: Double? = null

  @Column(name = "type", nullable = false, length = 32)
  var type: String? = null

  @Column(name = "refProduit", nullable = false)
  var refProduit: Int? = null

  @Column(name = "motif", nullable = false, length = 128)
  var motif: String? = null
}

File: backend_py/pharmaPython/app/models/ligne_caisse.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class LigneCaisse(Base):
  __tablename__ = "ligne_caisse"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  caisse_id = Column(Integer, ForeignKey("caisse.id"))
  produit_id = Column(Integer, ForeignKey("produit.id"))
  libelle = Column(String, nullable=False)
  date_ligne = Column(DateTime, nullable=False)
  debit = Column(Float, default=0.0)
  credit = Column(Float, default=0.0)
  type = Column(String(32), nullable=False)
  ref_produit = Column(Integer, nullable=False)
  motif = Column(String(128), nullable=False)

  caisse = relationship("Caisse", lazy="select")
  produit = relationship("Produit", lazy="select")
